### DISCUSSION TITLE
Error deleting streams (all as well as specific monitorId) while using the app HLS Stream Monitor

### DISCUSSION THREAD

**Discussion by anderswassen** [EyevinnOSC/community#40] [Q&A] [answered]
Hi!

I am using the app HLS Stream Monitor and I found it really useful. However, I have faced an issue deleting streams that I monitor. I can't delete monitored streams using:

/monitor | DELETE | Delete all monitored streams
or
/monitor/:monitorId | DELETE | Delete a specific stream

An example:
curl -X 'DELETE' \
  'https://awatenant-hlsmonitorawa.eyevinn-hls-monitor.auto.prod.osaas.io/monitor/20c07128-be3c-41fc-8253-a523ffbba19d' \
  -H 'accept: application/json'
results in:
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "Cannot read properties of null (reading 'streams')"
}

Any idea what I am doing wrong?

Keep up the great work!

**Comment by JoeldelPilar** [comment, marked as answer]
Hi @anderswassen 

This seems to be a bug in the HLS-monitor service. We will report it to the maintainer of the repository! 

Issue: https://github.com/Eyevinn/hls-monitor/issues/40

If there is any issue remaining just let us know and we help you as fast as we can! 

Regards / Joel del Pilar

**Reply by JoeldelPilar** [reply]
Ping! @anderswassen
 
https://github.com/Eyevinn/hls-monitor/issues/40#issuecomment-2403331871

I hope this might be helpful and solves the issue?

Regards / Joel del Pilar

**Reply by birme** [reply]
Hi! This issue has been addressed and the hls-monitor version in Open Source Cloud has been updated.

/Jonas and the OSC team

**Reply by anderswassen** [reply]
Hi!

Well, the documentation is still not updated nor swagger so I haven't tried to delete a specific stream from being monitored (as suggested in the comment).

There are two ways to delete streams from being monitored. If I understood the comment correctly, the first endpoint (DELETE /monitor/{monitorId}) also requires the stream value. Is that really needed? Can a "monitorId" monitor multiple streams? 

The second endpoint (DELETE /monitor) that should "Delete all monitored steams" requires  "monitorId": "string" as an object. May I suggest that that endpoint is updated to actually delete all monitored streams without the need of me adding both monitorId?

Cheers!

**Reply by anderswassen** [reply]
Hehe, I figured out how to monitor multiple streams using one monitorId. And I can confirm that the endpoint DELETE /monitor/{monitorId} works as expected.

The question regarding the second endpoint is still valid.

Thanks!

**Reply by birme** [reply]
Hi, You need to recreate your instance. I did not restart it so you are probably still running the old version.

The concept is that you have monitors and each monitor can have the role to monitor a set of streams. Every monitor is a single process so by creating multiple monitors you get multiple processes.

The reason why you provide the streams to DELETE /monitor/{monitorId} is to be able to remove one or many streams from that monitor at the same time. And to delete an actual monitor you do DELETE /monitor with {monitorId} in the body.

I agree that the API is not as REST ful as it should be and candidate for restructuring in a major version update of the API.

Great feedback and appreciated!

/Jonas

**Reply by anderswassen** [reply]
100% aligned! Realized that I forgot to restart my instance and it works as expected after that. I fully understand the reason for providing streams and monitorId as you explained above. Thanks for that great explanation.

Keep up the great work!
